### ISSUE TITLE
Change image size (Resize image)

### ISSUE THREAD

**Issue by SebZar** [d2phap/ImageGlass#1458]
### Is your feature request related to a problem? Please describe.
We used previously IrfanView and there you have a function to resize the image.
We miss this function.

### Describe the solution you'd like
Adding a new dialog/function for image resizing to change the size of the current image.
- new size in pixel x * y
- or new size in percent x%, y%
- Keep ratio proportional

**Comment by leo-smi** [comment]
Is this feature on the microsoft store version?

**Comment by d2phap** [comment]
> Is this feature on the microsoft store version?

No, IG Store currently has the same features as the Classic release.

**Comment by tdbe** [comment]
I've never been more confused in my entire life. So ImageGlass, has an advanced, Adobe-grade, Cropping tool:

![image](https://github.com/d2phap/ImageGlass/assets/1399607/5f3777f9-a18e-4c79-b8a7-f69028d787ba)

A tool which btw can re-encode and export to many advanced image formats.

but it CANNOT RESIZE AN IMAGE? Like, click button, enter resolution, click save? I thought I was missing something obvious. 🫠

Cannot imagine, what are the roadblocks for this feature?

**Comment by d2phap** [comment]
> what are the roadblocks for this feature?

It does not align to (or has low priority) ImageGlass vision: an image viewer, not editor. The Flip/Rotate/Crop features have just added lately after many requests, not from the initial roadmap.

**Comment by Timmmmaaahh** [comment]
- IrfanView: supports resizing
- FastStone: supports resizing
- XnView: supports resizing
- Picturereflect: supports resizing
- Honeyview: supports resizing
- JPEGView: supports resizing
- nomacs: supports resizing
I could go on for a while, but you probably get the point: it's quite unusual for a 3rd party image **viewer** to not support resizing (but does support cropping for some reason). It should be their #1 priority to be on par with all other applications. I personally use ImageGlass mainly for viewing, and have occasionally missed this feature. I just solved it with the Image Resizer PowerToy.

**Comment by tdbe** [comment]
Yeah good points. I also had to start installing and explaining PowerToys to someone after failing to resize an image.

Basically was trying to brag to my gf, who is used to macs, that "it just works with this awesome default image viewer" but it doesn't 😅

PS: But ok I'm aware it's more than 0 effort to add resizing; you need to set up a few different resizing rule options etc.. But at least we already have the multiple image format rasterization support.

**Comment by donatopace** [comment]
Similar to:
https://github.com/d2phap/ImageGlass/discussions/1830
with the addition of the resampling methods

**Comment by nicolafarina** [comment]
Cannot be more agree. It's an incredible missing feature!!!

**Comment by d2phap** [comment]
Implemented in [ImageGlass 9.2 Beta](https://github.com/d2phap/ImageGlass/releases/tag/9.1.8.1104). You can access Resize tool from Main Menu > Tools > Resize image, or press Alt+R.
- This tool utilizes https://photosauce.net library to scale the image
- Supported scaling range: from 1 pixel to 16,384 pixels
- Supports formats: single-frame non-animated formats
- Supports resizing Clipboard image: paste an image from clipboard and resize
- Works with Rotation, Flip features

After clicking "OK" button in the Resize tool, the image is scaled and updated on the viewer as "Clipboard image". It won't auto-save the image.

```
⚠️ Note:
   Similar to other editing features, saving resized image does not include its metadata such as exif profile.
```

![image](https://github.com/user-attachments/assets/3d136c93-0a73-41bf-9501-8238d232fb16)

**Comment by leo-smi** [comment]
> Implemented in [ImageGlass 9.2 Beta](https://github.com/d2phap/ImageGlass/releases/tag/9.1.8.1104). You can access Resize tool from Main Menu > Tools > Resize image, or press Alt+R.
> 
> * This tool utilizes https://photosauce.net library to scale the image
> * Supported scaling range: from 1 pixel to 16,384 pixels
> * Supports formats: single-frame non-animated formats
> * Supports resizing Clipboard image: paste an image from clipboard and resize
> * Works with Rotation, Flip features
> 
> After clicking "OK" button in the Resize tool, the image is scaled and updated on the viewer as "Clipboard image". It won't auto-save the image.
> 
> ```
> ⚠️ Note:
>    Similar to other editing features, saving resized image does not include its metadata such as exif profile.
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/3154213/382792534-3d136c93-0a73-41bf-9501-8238d232fb16.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4MTgzOTksIm5iZiI6MTczMDgxODA5OSwicGF0aCI6Ii8zMTU0MjEzLzM4Mjc5MjUzNC0zZDEzNmM5My0wYTczLTQxYmYtOTUwMS04MjM4ZDIzMmZiMTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDVUMTQ0ODE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2U2NWE5MjZiNjVmY2RjZmFiMzU3MGJhMDFhNTdiY2ZiNDcwYWUxNTJjODg0NjUyNmQ4ODM1MDQ1NDc0MDQ5MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.789EnqWA_MG_HQrTEkKA3Dg4CarhJpGIicJM3eqVg0c)

thank you!
